.circleci/config.yml

```diff
@@ -1,71 +1,68 @@
 version: 2
 jobs:
-  # Build and deploy- Just for master
-  build-and-deploy:
+  build:
     docker:
-      - image: circleci/node:lts
-      
-    working_directory: ~/repo
-
+      - image: circleci/buildpack-deps:buster
+    environment:
+        BUILD_IMAGE: italia/publiccode-editor-build
+        FINAL_IMAGE: italia/publiccode-editor
     steps:
       - checkout
-      # Run install 
-      - run: yarn install
-      # Run lint
+      - setup_remote_docker
+
+      - run: echo $DOCKER_PASSWORD | docker login --username italia --password-stdin
+
+      # Build the image with development tools and push it to Dockerhub
+      - run: docker build --target build-stage -t $BUILD_IMAGE:$CIRCLE_SHA1 .
+      - run: docker push $BUILD_IMAGE:$CIRCLE_SHA1
+
+      # Build the final image
+      - run: docker build -t $FINAL_IMAGE:$CIRCLE_SHA1 .
+      - run: docker push $FINAL_IMAGE:$CIRCLE_SHA1
+
+  test:
+    docker:
+        - image: italia/publiccode-editor-build:${CIRCLE_SHA1}
+    working_directory: /app
+    steps:
       - run: yarn lint
-      # Run test
       - run: yarn test
-      # Build 
-      - run: export ELASTIC_URL=$ELASTIC_URL_ENV && yarn build-prod 
-      # Check `rsync` 
+
+  deploy:
+    docker:
+      - image: circleci/buildpack-deps:buster
+    environment:
+        FINAL_IMAGE: italia/publiccode-editor
```

Comment:
The duplication here is unfortunately due to CircleCI not allowing global variables in the YAML, AFAIK.